src/main/webapp/index-script.js

```diff
@@ -383,24 +383,114 @@ function degreesToRadians(degrees) {
 }
 
 /* Fills out search results page. */
-function populateSearch(places) {
-  places = sortPlacesByRating(places);
+function populateSearch(places, location) {
+  const resultSortDistance = document.getElementById("sort-distance");
+  const resultSortRating = document.getElementById("sort-rated");
+
   triggerModal(document.getElementById("results-popup"));
   populateResults(places);
+
+  // Adds an event listener for the distance sort button
+  resultSortDistance.addEventListener("click", () => {
+    console.log("Clicked!!");
+    console.log(places);
+    const placePromise = new Promise((resolve, reject) => {
+      places = sortPlacesByDistance(places, location);
+      console.log("In the promise...");
+      // Timeout required to allow sortPlacesByDistance to finish.
+      setTimeout(function run() {resolve(places)}, 2000);
+    });
+    placePromise.then((places) => {
+      console.log("In the .then....");
+      console.log(places);
+      closeModal(document.getElementById("results-popup"));
+      triggerModal(document.getElementById("results-popup"));
+      populateResults(places);
+    });
+  });
+  
+  // Adds an event listener for the rating sort button
+  resultSortRating.addEventListener("click", () => { 
+    console.log("Clicked!!");
+    console.log(places);
+    const placePromise = new Promise((resolve, reject) => {
+      places = sortPlacesByRating(places, location);
+      console.log(places);
+      console.log("In the promise...");
+      // Timeout required to allow sortPlacesByRating to finish. 
+      setTimeout(function run() {resolve(places)}, 2000);
+    }).then((places) => {
+      console.log("In the .then....");
+      console.log(places);
+      closeModal(document.getElementById("results-popup"));
+      triggerModal(document.getElementById("results-popup"));
+      populateResults(places);
+    });
+  });
 }
 
-/* Adds a field 'rating' to each place with a random integer to
- * simulate sorting locations by their ratings. */
+/** Sorts place options by rating */
 function sortPlacesByRating(places) {
-  places.forEach((place) => {
-    let rand = Math.floor(Math.random() * 10);
-    place.rating = rand;
+  places.sort(function(a, b){
+    (getPlaceRating(a.place_id) > getPlaceRating(b.place_id)) ? 1 : -1});
```

Comment:
Right now seems like ties default to putting b first, is this how we want to break our ties? Was there a specific reason for this?

Reply:
I just modified the original statement that you (or dterr?) had written regarding the ratings to use the getPlaceRating method instead of .rating, so I figured you (or dterr?) wanted it this way!

Reply:
I did not write this. I believe it is an old stub we were using. I think arbitrarily breaking ties is good enough for where we are in our timeline.

Reply:
I was referrring specifically to how we break ties when two a.place_id == b.place_id, but I guess we're on a time crunch so this isn't that important